packages/quantic/force-app/main/default/lwc/quanticFeedbackModal/quanticFeedbackModal.js

```diff
@@ -0,0 +1,215 @@
+    yourResponsesHelp,
+    thankYouForYourFeedback,
+  };
+
+  /** @type {string} */
+  optionsValue;
+  /** @type {string} */
+  detailsValue;
+  /** @type {boolean} */
+  feedbackSubmitted = false;
+  /** @type {boolean} */
+  detailsInputError = false;
+  /** @type {string} */
+  error;
+  /** @type {boolean} */
+  validated = false;
+
+  connectedCallback() {
+    this.validateFeedbackModal();
+  }
+
+  /**
+   * Validates the properties provided to the feedback moda.
+   */
+  validateFeedbackModal() {
+    getBueno(this).then(() => {
+      this.validateOptionList();
+      if (!this.error) {
+        this.validateOptions();
+      }
+      this.validated = true;
+    });
+  }
+
+  validateOptionList() {
+    if (!Bueno.isArray(this.options)) {
+      console.error('The options provided are not in a valid array.');
+      this.setError();
+    } else if (!this.options.length) {
+      console.error('At least one option must be specified.');
+      this.setError();
+    }
+  }
+
+  validateOptions() {
+    const missingValueOrLabel = this.options?.some(
+      ({value, label}) => !value || !label
+    );
+    if (missingValueOrLabel) {
+      console.error(
+        'In the c-quantic-feedback-modal, each option requires a label and a value to be specified.'
+      );
+      this.setError();
+    } else {
+      this.options.forEach(({value, label}) => {
+        if (!Bueno.isString(value)) {
+          console.error(`The "${value}" value is not a valid string.`);
+          this.setError();
+        }
+        if (!Bueno.isString(label)) {
+          console.error(`The "${label}" label is not a valid string.`);
+          this.setError();
+        }
+      });
+    }
+  }
+
+  setError() {
+    this.error = 'c-quantic-feedback-modal Error';
+  }
+
+  /**
+   * Closes the modal.
+   * @returns {void}
+   */
+  closeModal() {
+    this.close();
+  }
+
+  /**
+   * Returns the lightning-radio-group element.
+   * @returns {{setCustomValidity: function, reportValidity: function}}
+   */
+  get optionsInput() {
+    return this.template.querySelector('lightning-radio-group');
+  }
+
+  /**
+   * Returns the CSS class of the reason input.
+   */
+  get detailsFormClass() {
+    return `slds-form-element ${
+      this.detailsInputError ? 'slds-has-error' : ''
+    }`;
+  }
+
+  /**
+   * Handles the option change.
+   * @param {CustomEvent} event
+   */
+  handleOptionChange(event) {
+    this.detailsInputError = false;
+    this.optionsInput.setCustomValidity();
+    this.optionsInput.reportValidity();
+    this.optionsValue = event.detail.value;
+  }
+
+  /**
+   * Handles the details input change.
+   * @param {CustomEvent} event
+   */
+  handleDetailsChange(event) {
+    this.detailsInputError = false;
+    this.detailsValue = event.target.value;
+  }
+
+  /**
+   * Indicates whether the details input should be displayed.
+   * @returns {boolean}
+   */
+  get displayDetailsInput() {
+    return this.optionsValue === 'other';
```

Comment:
Since the `other` option  has a special behavior, it should appear in the `options` documentation.

However, considering the options can be customized, what would you think of accepting these optional attributes when configuring the options: `withDetails`, `optionalDetails` instead of relying on a special option value?

It would allow an implementer to include details for more than one option and/or make the details optional if needed.

Reply:
Very neat idea! this would add a very useful freedom when using the component, will implement that 👌